Assets/Scripts/Models/Area/RoomBehavior.cs

```diff
@@ -387,16 +387,15 @@ public RoomBehavior Clone()
         public void Control(Room room)
         {
             this.Room = room;
-            List<Tile> innerTiles = room.GetInnerTiles();
-            List<Tile> borderTiles = room.GetBorderingTiles();
-
-            List<Tile> allTiles = innerTiles.Union(borderTiles).ToList();
+            HashSet<Tile> innerTiles = room.GetInnerTiles();
+            HashSet<Tile> borderTiles = room.GetBoundaryTiles();
+            innerTiles.UnionWith(borderTiles);
```

Comment:
But in reality, it won't improve any memory, just a small performance cost (very small)